### PR TITLE
Documentation added tutorials-evaluating-script-structure.rst

### PR DESCRIPTION
**Pull request by magdesign**
added on request in the [forum](https://discuss.pixls.us/t/call-for-volunteers-merge-http-opticalenquiry-com-nuke-into-natron-doc/24690) the section:
http://www.opticalenquiry.com/nuke/index.php?title=Evaluating_Script_Structure
from Martin Constable to the docs.
new file is: tutorials-evaluating-script-structure.rst
- added 3 new images
- added some  link targets for cross referencing
- added entry to show in toctree

### CODE REVIEW

**Comment by devernay** [comment]
Thank you!
I'll check this when I have time, unless someone else wants to review.
The build is available at: https://natron--612.org.readthedocs.build/en/612/

File: Documentation/source/guide/compositing-color.rst [hunk 1]
```diff
@@ -1,8 +1,103 @@
...
+
+
+
+The nodes
+---------
+Really important nodes are marked with an asterix:
+
+
+
+**Add****
```
Comment (devernay) [inline Documentation/source/guide/compositing-color.rst]: too many `*`

File: Documentation/source/guide/compositing-color.rst [hunk 2]
```diff
@@ -1,8 +1,103 @@
...
+
+This node affects all values within the image in the same way: literally adding to them.
+Positive values will brighten all parts of the image. As a stand alone color manipulation it is of limited use, though it is ideal for lightening the blacks of distant objects.
+
+
+**Multiply***
```
Comment (devernay) [inline Documentation/source/guide/compositing-color.rst:24]: too many `*`

File: Documentation/source/guide/compositing-color.rst [hunk 3]
```diff
@@ -1,8 +1,103 @@
...
+The saturation channel of HSL can, for example, have a contrast adjustment applied to it which could desaturate the less saturated parts of an image and super saturate the remainder. Woot! Try that in Photoshop!
+Lab color space is another useful fellow. It is the quantum physics of the color world and I shall but kiss the shadow of it's vast and complex form. The fascinating thing about this space is that it separates the lightness values of an image (the L channel) from it's hue and saturation (the a and b channel combined). HSL also does this but not nearly so well. You can take the a and b and move them into those of another image.
+The effect of this is similar to image editors hue blend mode. I have found it useful to augment the colorfulness of a dull sky by using the blurred color values from a vivid sunset.
+Try also blurring the a and b channels. This will blur only the hue and saturation components of an image and leave its lightness values alone.
+A novel use for the ColorSpace node can be found in the Assets page (see the Double Rainbows asset).
+
+**ColorCorrect** *
```
Comment (devernay) [inline Documentation/source/guide/compositing-color.rst]: dangling `*`

File: Documentation/source/guide/compositing-color.rst [hunk 4]
```diff
@@ -1,8 +1,103 @@
...
+
+Artists have for thousands of years been separating the lightness values of their paintings into three bands: shadows, midtones and highlights. The ColorCorrect node is a collection of operations that not only can effect the entire image but can address separately these three ranges 
+
+**Gamma**
+
+This raises or lowers the middle-ish point of the color curve. The default value is one, with smaller numbers darkening the lower registers and higher numbers lightening them. There is a Gamma node but I find that stand-alone gamma adjustments are best done using ColorLookup so as to give you flexibility over where the ‘grab point‘ of your curve is. Both ColorCorrect and Grade have built in Gamma value sliders.
+
+**Grade** *
```
Comment (devernay) [inline Documentation/source/guide/compositing-color.rst]: `*`

File: Documentation/source/guide/compositing-color.rst [hunk 5]
```diff
@@ -1,8 +1,103 @@
...
+
+This node is a collection of operations that combine to work upon the lightness and hue values of an image. It is mostly a fixer: used for correcting and matching, though of course it can also be used for more aesthetically lavish purposes.
+
+**Histogram**
+
+**HueCorrect** *
```
Comment (devernay) [inline Documentation/source/guide/compositing-color.rst]: `*`

File: Documentation/source/guide/compositing-color.rst [hunk 6]
```diff
@@ -1,8 +1,103 @@
...
+
+This can be a very tricky node to get to know.
+We can conclude that the perception of the amount of hue within a color (its chroma) is linked to two things: saturation and lightness.
+When adjusting color it is important to have separate control over these values, which is something that the HueCorrect node offers.
+This effect it masks by two further values: hue and saturation. Its interface offers control over nine values:
+
+Saturation (sat)
```
Comment (devernay) [inline Documentation/source/guide/compositing-color.rst:73]: format as title?

File: Documentation/source/guide/tutorials-evaluating-script-structure.rst [hunk 7]
```diff
@@ -0,0 +1,208 @@
+.. for help on writing/extending this file, see the reStructuredText cheatsheet
+   http://github.com/ralsina/rst-cheatsheet/raw/master/rst-cheatsheet.pdf
+   
+   .. _evaluating-script-structure:
+
+
+Thanks to Martin Constable to allow us to use his documents.
```
Comment (devernay) [inline Documentation/source/guide/tutorials-evaluating-script-structure.rst]: Would add Martin (and you) to the Documentation authors (https://github.com/NatronGitHub/Natron/blob/RB-2.4/CONTRIBUTORS.txt and in the intro)

**Comment by magdesign** [comment]
- Fixed the issues with asterisk * signs
- Added name to CONTRIBUTORS.txt
- "working with color" was always there as title in prev. versions, left it like this

**Review by devernay — APPROVED** [review]

**Pull request merged**